lib/pwdutils.c

```diff
@@ -127,6 +128,26 @@ char *xgetlogin(void)
 	return NULL;
 }
 
+/*
+ * Return a pointer to a `struct group` for a matching group name or GID.
+ */
+struct group *ul_get_grp_by_name_or_id(const char *str) {
+	int rc = 0;
+	int64_t gid;
+	struct group *gr;
+
+	rc = ul_strtos64(str, &gid, 10);
+	if (((gid && gid < 0) || (gid && gid > SINT_MAX(gid_t))))
```

Comment:
The `gid` conditionals are unnecessarily complex.
If `gid < 0` or `gid > SINT_MAX()` then `gid` is never `0`.